tests/integration/test_atleast_nd.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+
+import cunumeric as num
+
+
+def _check(a, routine, sizes):
+    b = getattr(np, routine)(*a)
+    c = getattr(num, routine)(*a)
+    is_equal = True
+    err_arr = [b, c]
+
+    if len(b) != len(c):
+        is_equal = False
+        err_arr = [b, c]
+    else:
+        for each in zip(b, c):
+            if not np.array_equal(*each):
+                err_arr = each
+                is_equal = False
+                break
+    print_msg = f"np.{routine}({sizes})"
+    assert is_equal, (
+        f"Failed, {print_msg}\n"
+        f"numpy result: {err_arr[0]}\n"
+        f"cunumeric_result: {err_arr[1]}\n"
+        f"cunumeric and numpy shows"
+        f" different result\n"
+    )
+    print(f"Passed, {print_msg}, np: {b}" f", cunumeric: {c}")
+
+
+DIM = 10
+
+SIZE_CASES = [
+    (0,),  # empty array
+    (1,),  # singlton array
+    (10,),  # scalar
+    (DIM, 1),  # 1D array
+    (DIM, DIM),  # 2D array
+    (DIM, DIM, DIM),  # 3D array
+    (DIM, DIM, DIM, DIM),  # 4D array
```

Comment:
Note that the maximum number of dimensions is configurable. I suggest you follow the idiom used in other tests. (here's one example: https://github.com/nv-legate/cunumeric/blob/ac77bd955633d72af47196f732aad20a4edabfe1/tests/integration/test_logical.py#L53-L57)